src/main/resources/avro/common.avdl

```diff
@@ -2,6 +2,325 @@
 /**
 This protocol defines common types used in the other GA4GH protocols. It does
 not have any methods; it is merely a library of types.
+
+##Sequences, Segments, and References
+
+A sequence is a component in a sequence graph, optionally joined at its ends to
+other sequences. Sequences are not directly retrievable through the API, but
+ranges of their bases can be downloaded.
+
+A `Segment` is a range on a sequence, possibly including the joins at the
+sequence's ends. It does not include base data. When you ask about the novel
+sequences that come in a `VariantSet`, you will get back `Segment`s covering
+them. A `Path` through a sequence graph is a list of `Segment`s.
+
+A `Reference` (defined in references.avdl) is an object that marks a sequence as
+part of a `ReferenceSet`'s sequence graph. It contains a `Segment` covering the
+sequence it belongs to.
+
+##Sequence Graph Concepts
+
+Many of the types defined here are related to defining sequence graphs. Sequence
+graph concepts can be illustrated on a piece of double-stranded DNA with two
+nicks in it, one on the top strand and one on the bottom strand. Here is an
+example where the top strand is GGTGGNG.
+
+```
+            | <- top strand nick at position (2,+)
+5' -------  -------------------  3'
+     G   G   T   G   G   N   G
+     C   C   A   C   C   N   C
+3' ----------------  ----------  5'
+                  | <- bottom strand nick at position (3,-)
+
+    +0- +1- +2- +3- +4- +5- +6- <- coordinates
+```
+
+A sequence is a piece of double-stranded DNA composed of a series of DNA
+basepairs. In the default forward orientation a sequence is specified by the DNA
+letters of its top strand, e.g. in this case GGTGGNG, where N indicates an
+unknown base. The basepairs are indexed left to right from 0 to 6 relative to
+this default orientation.  A sequence containing regions of uncertainty, denoted
+by Ns, is called a scaffold, while a sequence in which every base is known is
+called a contig. This sequence is a scaffold.
+
+In its default forward orientation, each basepair has a left or "+" side and a
+right or "-" side. For example, the left side of the T/A basepair in the above
+example lies immediately to the left of this basepair. The left side of the T/A
+basepair is represented by the position (2,+), where the index 2 is the same as
+that of the basepair itself, and the "+" confirms that we mean the default
+forward orientation. The right side is indicated using the "-" orientation. For
+example, the right side of the following G/C base pair is represented by the
+position (3,-).
```

Comment:
An slightly confusing coincidence in this example is that, in this sequence of length 7, the index `3` could be interpreted as being measured from either end. 

The asymmetry between the `start` index having the nick come just *before* it while the `end` index has the nick *after* it (when both are read from left-to-right, as the indices are apparently intended to be read) made it hard for me to figure out all that was meant to be implied by the `(3,-)` notation. 

Let me know if that doesn't make sense.